### PR TITLE
Write in process stdin

### PR DESCRIPTION
**Pull request by crgimenes**
Relating to issue #21

Implements the possibility of writing in the container stdin as described in the issue.

### CODE REVIEW

File: provision/docker.go [hunk 1]
```diff
@@ -140,16 +142,36 @@ func FnKillContainer(client *docker.Client, containerID string) (err error) {
 func FnRun(client *docker.Client, containerID string) (Stdout *bytes.Buffer) {
 	err := client.StartContainer(containerID, nil)
 	if err != nil {
-		log.Println(err)
+		panic(err)
```
Comment (marioidival) [inline provision/docker.go]: [don't use panic](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic)

File: provision/docker.go [hunk 2]
```diff
@@ -140,16 +142,36 @@ func FnKillContainer(client *docker.Client, containerID string) (err error) {
...
+		Stdin:       true,
+		InputStream: strings.NewReader(Input),
+	})
+	if err != nil {
+		panic(err)
+	}
+
+	_, err = client.WaitContainer(containerID)
+	if err != nil {
+		panic(err)
```
Comment (marioidival) [inline provision/docker.go]: [don't use panic](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic)

File: provision/docker.go [hunk 3]
```diff
@@ -140,16 +142,36 @@ func FnKillContainer(client *docker.Client, containerID string) (err error) {
...
 	stdout := new(bytes.Buffer)
 
-	client.Logs(docker.LogsOptions{
+	err = client.Logs(docker.LogsOptions{
 		Container:    containerID,
 		Stdout:       true,
 		OutputStream: stdout,
 	})
+	if err != nil {
+		panic(err)
```
Comment (marioidival) [inline provision/docker.go]: [don't use panic](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic)

File: provision/docker.go [hunk 4]
```diff
@@ -140,16 +142,36 @@ func FnKillContainer(client *docker.Client, containerID string) (err error) {
...
+
+	_, err = client.AttachToContainerNonBlocking(docker.AttachToContainerOptions{
+		Container:   containerID,
+		RawTerminal: true,
+		Stream:      true,
+		Stdin:       true,
+		InputStream: strings.NewReader(Input),
+	})
+	if err != nil {
+		panic(err)
```
Comment (marioidival) [inline provision/docker.go]: [don't use panic](https://github.com/golang/go/wiki/CodeReviewComments#dont-panic)

**Comment by avelino** [comment]
@cassiobotaro review pls

File: examples/testDocker/main.py [hunk 5]
```diff
@@ -3,10 +3,13 @@
 import json
 from random import randint
 import time
+import sys
+
+data = sys.stdin.readlines() # read stdin
```
Comment (cassiobotaro) [inline examples/testDocker/main.py]: put one more space before comments, keep thee pep 8

File: gofn.go [hunk 6]
```diff
@@ -1,10 +1,15 @@
...
 
 import (
+	"bytes"
+
 	docker "github.com/fsouza/go-dockerclient"
 	"github.com/nuveo/gofn/provision"
 )
 
+// Input receives a string that will be written to the stdin of the container
+var Input string
```
Comment (cassiobotaro) [inline gofn.go]: Why is this variable global?

Comment (crgimenes) [inline gofn.go]: as specified in issue.

**Pull request merged**